### PR TITLE
Add Onnx Config for PoolFormer

### PR DESCRIPTION
**Pull request by BakingBrains**
# What does this PR do?

Fixes # (https://github.com/huggingface/transformers/issues/16308)

Add changes to make PoolFormer models available for Onnx conversion.

Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ChainYo

### CODE REVIEW

**Comment by BakingBrains** [comment]
I have ran 
```RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "poolformer"```

tt

**Comment by BakingBrains** [comment]
Conversion output

tt2

**Comment by BakingBrains** [comment]
@ChainYo @lewtun Any suggestions here?

Thanks and Regards.

File: src/transformers/onnx/features.py [hunk 1]
```diff
@@ -52,7 +50,7 @@
 
 
 def supported_features_mapping(
-    *supported_features: str, onnx_config_cls: str = None
+        *supported_features: str, onnx_config_cls: str = None
```
Comment (ChainYo) [inline src/transformers/onnx/features.py:53]: Extra whitespace.

File: src/transformers/onnx/features.py [hunk 2]
```diff
@@ -523,7 +523,7 @@ class FeaturesManager:
 
     @staticmethod
     def get_supported_features_for_model_type(
-        model_type: str, model_name: Optional[str] = None
+            model_type: str, model_name: Optional[str] = None
```
Comment (ChainYo) [inline src/transformers/onnx/features.py:526]: Extra whitespace

File: src/transformers/models/poolformer/configuration_poolformer.py [hunk 3]
```diff
@@ -125,3 +130,20 @@ def __init__(
...
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("pixel_values", {0: "batch", 1: "num_channels", 2: "height", 3: "width"}),
+            ]
+        )
+
+    @property
+    def atol_for_validation(self) -> float:
+        return 1e-2
```
Comment (ChainYo) [inline src/transformers/models/poolformer/configuration_poolformer.py:149]: `1e-2` looks low for `atol`. I won't go more than `1e-3`, even if `1e-5` is best.

File: docs/source/en/serialization.mdx [hunk 4]
```diff
@@ -74,7 +74,6 @@ Ready-made configurations include the following architectures:
 - GPT-J
 - GroupViT
 - I-BERT
-- ImageGPT
```
Comment (ChainYo) [inline docs/source/en/serialization.mdx:77]: Is it intended to remove `ImageGPT`?

**Review by chainyo — CHANGES_REQUESTED** [review]
Thanks for working on this!

Don't forget to use `$ make fixup` command before pushing your work.

File: docs/source/en/serialization.mdx (hunk 4 above)
Comment (BakingBrains) [reply docs/source/en/serialization.mdx]: sorry, I think I merged older version of main. It's fixed.😅

File: src/transformers/models/poolformer/configuration_poolformer.py (hunk 3 above)
Comment (BakingBrains) [reply src/transformers/models/poolformer/configuration_poolformer.py]: @ChainYo Understood, but if I set it anywhere between ```1e-3``` to ```1e-5```, then output values doesn't match while validating. Any suggestions?

Thank you

Comment (BakingBrains) [reply src/transformers/models/poolformer/configuration_poolformer.py]: The outputs match on ```2e-3```

Comment (BakingBrains) [reply src/transformers/models/poolformer/configuration_poolformer.py]: image

Comment (ChainYo) [reply src/transformers/models/poolformer/configuration_poolformer.py]: Cool, I think `2e-3` is okay! Let's see how @lewtun feels about that.

Comment (BakingBrains) [reply src/transformers/models/poolformer/configuration_poolformer.py]: @lewtun ?

Comment (ChainYo) [reply src/transformers/models/poolformer/configuration_poolformer.py]: Please do not multiple pings. It doesn't help :hugs:

Comment (BakingBrains) [reply src/transformers/models/poolformer/configuration_poolformer.py]: Sorry😅

Comment (ChainYo) [reply src/transformers/models/poolformer/configuration_poolformer.py]: Also, could you fix the linting for CI?

**Comment by BakingBrains** [comment]
Any updates on the PR?

**Comment by BakingBrains** [comment]
@ChainYo Any updates here?

**Comment by ChainYo** [comment]
Hey @lewtun and @michaelbenayoun, we should merge this before the Optimum change. 

Could you help on this?

**Review by michaelbenayoun — APPROVED** [review]
LGTM (after resolving conflicts)

**Comment by BakingBrains** [comment]
@michaelbenayoun Sure, I will add it to Optimum. 

Thank you

**Comment by BakingBrains** [comment]
I tried to resolve the conflicts, but I think I messed up

**Comment by BakingBrains** [comment]
I just reopened a new pull request for the same, can you please check @michaelbenayoun

**Pull request closed without merging**